homeassistant/components/metoffice/sensor.py

```diff
@@ -30,175 +32,163 @@
 ATTR_SITE_ID = "site_id"
 ATTR_SITE_NAME = "site_name"
 
-ATTRIBUTION = "Data provided by the Met Office"
-
-CONDITION_CLASSES = {
-    "cloudy": ["7", "8"],
-    "fog": ["5", "6"],
-    "hail": ["19", "20", "21"],
-    "lightning": ["30"],
-    "lightning-rainy": ["28", "29"],
-    "partlycloudy": ["2", "3"],
-    "pouring": ["13", "14", "15"],
-    "rainy": ["9", "10", "11", "12"],
-    "snowy": ["22", "23", "24", "25", "26", "27"],
-    "snowy-rainy": ["16", "17", "18"],
-    "sunny": ["0", "1"],
-    "windy": [],
-    "windy-variant": [],
-    "exceptional": [],
-}
-
-DEFAULT_NAME = "Met Office"
-
-VISIBILITY_CLASSES = {
-    "VP": "<1",
-    "PO": "1-4",
-    "MO": "4-10",
-    "GO": "10-20",
-    "VG": "20-40",
-    "EX": ">40",
-}
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=35)
-
-# Sensor types are defined like: Name, units
+# Sensor types are defined as: variable -> title, units, icon, enabled by default
 SENSOR_TYPES = {
-    "name": ["Station Name", None],
-    "weather": ["Weather", None],
-    "temperature": ["Temperature", TEMP_CELSIUS],
-    "feels_like_temperature": ["Feels Like Temperature", TEMP_CELSIUS],
-    "wind_speed": ["Wind Speed", SPEED_MILES_PER_HOUR],
-    "wind_direction": ["Wind Direction", None],
-    "wind_gust": ["Wind Gust", SPEED_MILES_PER_HOUR],
-    "visibility": ["Visibility", None],
-    "visibility_distance": ["Visibility Distance", LENGTH_KILOMETERS],
-    "uv": ["UV", UV_INDEX],
-    "precipitation": ["Probability of Precipitation", UNIT_PERCENTAGE],
-    "humidity": ["Humidity", UNIT_PERCENTAGE],
+    "name": ["Station Name", None, "mdi:label-outline", False],
+    "weather": [
+        "Weather",
+        None,
+        "mdi:weather-sunny",  # but will adapt to current conditions
+        True,
+    ],
+    "temperature": ["Temperature", TEMP_CELSIUS, "mdi:thermometer", True],
```

Comment:
Temperature sensors should use that device class.

Entities that set a device class should not set a custom icon.

Please set device class for as many entities as possible:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Honestly not sure what you mean about this. Where do I set the "device class" in the sensor? Each has name/state/unit/icon attributes - where does the "device class" fit into this, please?

Reply:
Please implement the `device_class` property in the sensor entity class.

Reply:
Ah right, I see - okay, on it.

Reply:
Done, thanks for that.